backend/api/Areas/Tools/Helpers/ImportProjectsHelper.cs

```diff
@@ -203,14 +300,100 @@ private Entity.Project Merge(Entity.Project project, Model.ImportProjectModel mo
             }
         }
 
+        /// <summary>
+        /// Append a new note to the project.
+        /// Parse the 'noteType' value, or set it as 'General'.
+        /// </summary>
+        /// <param name="project"></param>
+        /// <param name="noteType"></param>
+        /// <param name="note"></param>
+        private void AddNote(Entity.Project project, string noteType, string note)
+        {
+            if (!String.IsNullOrWhiteSpace(note))
+            {
+                if (!Enum.TryParse(noteType, true, out Entity.NoteTypes type))
+                {
+                    type = noteType switch
+                    {
+                        "Property" => Entity.NoteTypes.Public,
+                        "PID" => Entity.NoteTypes.Public,
+                        "LandLegalDescription" => Entity.NoteTypes.Public,
+                        "Surplus Dec / Readiness Checklist" => Entity.NoteTypes.Public,
+                        "TBL - CBA" => Entity.NoteTypes.Public,
+                        "Enhanced Referral Comments" => Entity.NoteTypes.Private,
+                        "Interest from Enhanced Referral" => Entity.NoteTypes.AgencyInterest,
+                        "Business Case Comments" => Entity.NoteTypes.Private,
+                        "Business Case Submitted" => Entity.NoteTypes.Private,
+                        "Clearance Letter Issued" => Entity.NoteTypes.Private,
+                        "Appraisal Date" => Entity.NoteTypes.Appraisal,
+                        "Assessed & Appraised Comments" => Entity.NoteTypes.Appraisal,
+                        "Weekly Review" => Entity.NoteTypes.Private,
+                        _ => Entity.NoteTypes.General
+                    };
+                    note = $"{noteType}{Environment.NewLine}{note}";
+                }
+
+                var projectNote = project.Notes.FirstOrDefault(n => n.NoteType == type);
+                if (projectNote == null)
+                {
+                    project.Notes.Add(new Entity.ProjectNote(project, type, note));
+                }
+                else
+                {
+                    if (!String.IsNullOrWhiteSpace(projectNote.Note))
+                        projectNote.Note += $"{Environment.NewLine}{Environment.NewLine}{note}";
+                    else
+                        projectNote.Note = note;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Find the property for the specified 'pid' in inventory and add it to the project.
+        /// Assumption is made that if the parcel is added, then also add all the buildings on the parcel.
+        /// </summary>
+        /// <param name="project"></param>
+        /// <param name="pid"></param>
+        /// <returns></returns>
+        private Entity.Project AddProperty(Entity.Project project, string pid)
+        {
+            try
+            {
+                var pidValue = Int32.Parse(pid.Replace("-", ""));
```

Comment:
.net used to throw a warning unless you specified the desired base here, I assume it doesn't?

Reply:
No warnings are thrown here, it is strongly typed at this point.  The only possible issue is that the `pid` is null, or results in not a number.  But I'm hesitant to add the logic at this point in time due to the current use-case for this helper.